k8s-scheduler/src/test/java/com/vmware/dcm/ScopeTest.java

```diff
@@ -0,0 +1,263 @@
+     */
+    @Test
+    public void testScopedSchedulerNodeLabels() {
+        final DBConnectionPool dbConnectionPool = new DBConnectionPool();
+        final DSLContext conn = dbConnectionPool.getConnectionToDb();
+        final List<String> policies = Policies.getDefaultPolicies();
+        final NodeResourceEventHandler nodeResourceEventHandler = new NodeResourceEventHandler(dbConnectionPool);
+        final int numNodes = 1000;
+        final int numPods = 20;
+        conn.insertInto(Tables.BATCH_SIZE).values(numPods).execute();
+
+        final PodEventsToDatabase eventHandler = new PodEventsToDatabase(dbConnectionPool);
+        final PodResourceEventHandler handler = new PodResourceEventHandler(eventHandler::handle);
+
+        for (int i = 0; i < numNodes; i++) {
+            final String nodeName = "n" + i;
+            final Map<String, String> nodeLabels = new HashMap<>();
+            if (i < 2) { // gpu: nodes 0..1
+                nodeLabels.put("gpu", "true");
+            }
+            if (i > 0 && i < 16) { // ssd: nodes 1..15
+                nodeLabels.put("ssd", "true");
+            }
+            final Node node = newNode(nodeName, nodeLabels, Collections.emptyList());
+            // more spare capacity as i increases
+            node.getStatus().getCapacity().put("cpu", new Quantity(String.valueOf(10 + 10 * i / numNodes)));
+            node.getStatus().getCapacity().put("memory", new Quantity(String.valueOf(1000 + 1000 * i / numNodes)));
+            node.getStatus().getCapacity().put("pods", new Quantity(String.valueOf(100)));
+            nodeResourceEventHandler.onAddSync(node);
+        }
+
+        for (int i = 0; i < numPods; i++) {
+            final String podName = "p" + i;
+            final Map<String, String> selectorLabels = new HashMap<>();
+            if (i < 8) { // pods requiring gpu: 0..7
+                selectorLabels.put("gpu", "true");
+            }
+            if (i > 1 && i < 10) { // pods requiring ssd: 2..9
+                selectorLabels.put("ssd", "true");
+            }
+            final Pod newPod = newPod(podName, UUID.randomUUID(), "Pending", selectorLabels, Collections.emptyMap());
+            handler.onAddSync(newPod);
+        }
+
+        final Scheduler scheduler = new Scheduler(dbConnectionPool, policies, "ORTOOLS", true, NUM_THREADS);
+        scheduler.setScopeOn();
+        scheduler.scheduleAllPendingPods(new EmulatedPodToNodeBinder(dbConnectionPool));
+
+        // Check that all pods have been scheduled to a node eligible by the scope filtering
+        final Result<PodInfoRecord> fetch = conn.selectFrom(Tables.POD_INFO).fetch();
+        assertEquals(numPods, fetch.size());
+        fetch.forEach(e -> assertTrue(e.getNodeName() != null
+                && (Integer.parseInt(e.getPodName().substring(1)) >= 10
+                  || Integer.parseInt(e.getNodeName().substring(1)) < 16)));
+    }
+
+
+    private Model buildModel(final DSLContext conn) {
```

Comment:
Are the below the methods the same as those in SchedulerTest? If so, just refer to them directly (and change to package-private visibility as needed).